jupyter_core/application.py

```diff
@@ -216,14 +234,22 @@ def start(self):
         """Start the whole thing"""
         if self.subcommand:
             os.execv(self.subcommand, [self.subcommand] + self.argv[1:])
-            return
+            raise NoStart()
```

Comment:
I think anything after execv is unreachable; if it succeeds, this code will no longer be executing, and if it fails, it will raise an error anyway.